proxy/coco/src/shared.rs

```diff
@@ -0,0 +1,29 @@
+use std::{ops::Deref, sync::Arc};
+
+use tokio::sync::RwLock;
+
+pub struct Shared<T> {
```

Comment:
We've come full circle.

Reply:
Hahaha ya, but this time I know better tricks

Reply:
Hubris!

Reply:
:smiling_imp: